test_functions/test_funcs.py

```diff
@@ -99,28 +101,42 @@ class TestFunction(object):
 
     __metaclass__ = ABCMeta
 
-    def __init__(self, dim):
+    def __init__(self, dim, verify=True):
         assert dim > 0
         self.dim = dim
+        self.verify = verify
         self.num_evals = 0
         self.min_loc = None
         self.fmin = None
         self.fmax = None
         self.bounds = None
         self.classifiers = []
 
+        # Note(Mike) - Not using the records yet, but will be soon
+        self.records = None
+        self.reset_records()
+
     def __repr__(self):
         return '{0}({1})'.format(self.__class__.__name__, self.dim)
 
     def evaluate(self, x):
-        x = numpy.asarray(x)
-        if x.shape != (self.dim,):
-          raise ValueError('Argument must be a numpy array of length {}'.format(self.dim))
+        if self.verify and (not isinstance(x, numpy.ndarray) or x.shape != (self.dim,)):
+            raise ValueError('Argument must be a numpy array of length {}'.format(self.dim))
 
         self.num_evals += 1
         value = self.do_evaluate(x)
+        to_be_returned = value.item() if hasattr(value, 'item') else value
+        self.update_records(now(), x, to_be_returned)
```

Comment:
why are we using `timeit` for this? why not just `time.time()` or `datetime.now()`?